lib/xcodeproj/project/project_helper.rb

```diff
@@ -157,6 +157,15 @@ def self.configuration_list(project, platform, deployment_target = nil, target_p
 
         cl.build_configurations << release_conf
         cl.build_configurations << debug_conf
+
+        project.build_configurations.each do |configuration|
+          next if %w(Debug Release).include?(configuration.name)
```

Comment:
Seems like build configurations can only be added one-by-one, though, to the `XCConfigurationList`- probably deliberately so.